### DISCUSSION TITLE
Contributions

### DISCUSSION THREAD

**Discussion by venaturum** [coin-or/pulp#386] [Q&A]
Hi devs, 

Although this is an open-source project, it is not clear whether contributions are welcome.

The only hint that they are is the fact that some issues in the bug tracker have been labelled "good first issue".

If I can sidestep this topic momentarily...

One of the recent issues labelled "good first issue" is https://github.com/coin-or/pulp/issues/328 which reports the search functionality in the online docs as broken.  To look into this, I built the docs locally and ran into another issue where the search failed to complete - the same as logged here https://github.com/sphinx-doc/sphinx/issues/8591.  That issue was caused by an update in sphinx, and was rectified by a corresponding update in sphinx_rtd_theme.  Note that I do not experience this issue with the online documentation, but then again it is not clear what version of sphinx is being used to build that (none of the dev requirements for this project are specified).

The docs for pulp use a rather obscure theme in sphinx_glpi_theme which is based on sphinx_rtd_theme.  I'd say the glpi theme needs a similar update to rectify the issue.  I swapped the glpi theme out for the rtd theme and built the docs and experience no issues.  

It is not clear to me what benefit the glpi theme is providing, especially in the context of having little support behind it.

Back to the original question... Are contributions wanted?  There are plenty of straightforward improvements that could be made to this project, in particular around attracting contributions (eg docs for contributing, dev environments, using pyproject.toml)

**Comment by pchtsp** [comment]
Hello and thanks for writing : )

Regarding your **original question**. Well, we do have a [contributing file](https://github.com/coin-or/pulp/blob/master/.github/CONTRIBUTING.md). If you believe the contributing guidelines / developer docs need polishing (or creation), feel free to propose changes through the discussions and then when agreed, with a PR.

As for the **theme question**: I don't really remember the justification for the present theme. I know that my requirements were: (1) a left menu bar that remains fixed when scrolling down and (2) it should look reasonably well. If you believe an improvement can be obtained with another theme, feel free to do a PR. Specially if you fix the docs issue!

Cheers,

F.

**Reply by venaturum** [reply]
Thanks Franco, I'll spin up a few ideas in the issues tracker over the next few days, and see what you think.
